### DISCUSSION TITLE
Finding Channel Management tedious. Can we use the default Channel Implementation in @grpc/grpc-js?

### DISCUSSION THREAD

**Discussion by akash931127** [deeplay-io/nice-grpc#701] [Q&A]
Hi, I was going through the code because I am finding managing channels among clients very tedious. I was trying to see if there was a default Channel Management already present in the underlying @grpc/grpc-js library.

I went through the following path:
- https://github.com/deeplay-io/nice-grpc/blob/master/packages/nice-grpc/src/client/ClientFactory.ts#L99-L102
- https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/make-client.ts#L129
- https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/client.ts#L125-L164

I am trying to avoid taking on the channel management as much as possible. What are the communities thoughts here? Should we provide the ability to use the default implementations available the underlying @grpc/grpc-js?

**Comment by aikoven** [comment]
Hi! Thanks for raising this. Could you clarify a bit more about the issue you’re facing with channel management? Are you looking for a way to automatically manage channels without manually creating or reusing them? Or is there something specific about the current behavior in @grpc/grpc-js or nice-grpc that isn’t working for your use case?

If you can share a specific example or a bit more detail, it will help the community understand and suggest better solutions.
